### PR TITLE
#8 logic and testing done

### PR DESCRIPTION
**Pull request by JavierGarna**
I've repeated the same process as #3 but adding a comment_count and ordering by created_at in SQL.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -145,4 +146,31 @@ describe("GET /api/users", () => {
...
+                        owner: expect.any(String),
+                        title: expect.any(String),
+                        review_id: expect.any(Number),
+                        category: expect.any(String),
+                        review_img_url: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        comment_count: expect.any(Number)
+                    })
+                )
```
Comment (bainesface) [inline __tests__/app.test.js:181]: Hey Javier!

You should have written another test to implement the additional comment count behaviour. When following TDD practice we do not want to change our previous, passing tests as we want them to still pass even with some extra functionality.

Please could you revert your original 200 test for 'GET api/reviews' and then add another test to handle the additional comment_count behaviour

File: app.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const express = require("express");
 const app = express();
-const { getCategories, getReview, patchReview, getUsers } = require("./controllers/controllers")
+const { getCategories, getReview, patchReview, getUsers, getReviews } = require("./controllers/controllers")
```
Comment (bainesface) [inline app.js:6]: You have one controllers file and one models file for all the requests. This isn't following the MVC pattern effectively and you should have a different file to handle the different types of requests. For example one each for review, comments, categories and users. 

I like to think that if some data has its own table in the database, then it should also have its own model and controller files

File: package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "husky": "^7.0.0",
     "jest": "^27.5.1",
     "jest-extended": "^2.0.0",
+    "jest-sorted": "^1.0.14",
```
Comment (emilyb93) [inline package.json:28]: I would like you to move any jest packages further down into your devDependencies object. This section is only for packages that are needed for the app to run when in production. As we wont be testing in production, we dont want these packages to be part of the final build.

**Review by emilyb93 — APPROVED** [review]
Thanks for refactoring and moving it all into separate MVC files, just a small change to make in your package.json and then feel free to merge :)

**Pull request merged**